src/udev/scsi_id/scsi_id.c

```diff
@@ -59,11 +60,9 @@ static char type_str[16];
 
 static void set_type(const char *from, char *to, size_t len) {
         int type_num;
-        char *eptr;
         const char *type = "generic";
 
-        type_num = strtoul(from, &eptr, 0);
-        if (eptr != from) {
+        if (safe_atoi(from, &type_num) >= 0) {
```

Comment:
See `scsi_std_inquiry()` in scsi_serial.c.
Please use `safe_atou_full(from, 16, &type_num);` and change the type of `type_num` from `int` to `unsigned`.

Reply:
Done.